wazo_dird_client/commands/helpers/base_source_command.py

```diff
@@ -2,61 +2,38 @@
 # Copyright 2019 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-from xivo_lib_rest_client import RESTCommand
+from wazo_dird_client.commands.helpers import DirdRESTCommand
 
 
-class SourceCommand(RESTCommand):
-
-    _rw_headers = {'Accept': 'application/json', 'Content-Type': 'application/json'}
-    _ro_headers = {'Accept': 'application/json'}
-
-    def create(self, body, tenant_uuid=None):
-        headers = dict(self._rw_headers)
-        tenant_uuid = tenant_uuid or self._client.tenant()
-        if tenant_uuid:
-            headers['Wazo-Tenant'] = tenant_uuid
+class SourceCommand(DirdRESTCommand):
 
+    def create(self, body, tenant_uuid=None, token=None):
+        headers = self.build_rw_headers(tenant_uuid, token)
         r = self.session.post(self.base_url, json=body, headers=headers)
         self.raise_from_response(r)
         return r.json()
 
-    def delete(self, source_uuid, tenant_uuid=None):
-        headers = dict(self._ro_headers)
-        tenant_uuid = tenant_uuid or self._client.tenant()
-        if tenant_uuid:
-            headers['Wazo-Tenant'] = tenant_uuid
-
+    def delete(self, source_uuid, tenant_uuid=None, token=None):
+        headers = self.build_ro_headers(tenant_uuid, token)
         url = '/'.join([self.base_url, source_uuid])
         r = self.session.delete(url, headers=headers)
         self.raise_from_response(r)
 
-    def get(self, source_uuid, tenant_uuid=None):
-        headers = dict(self._ro_headers)
-        tenant_uuid = tenant_uuid or self._client.tenant()
-        if tenant_uuid:
-            headers['Wazo-Tenant'] = tenant_uuid
-
+    def get(self, source_uuid, token=None, tenant_uuid=None):
```

Comment:
Style: all other methods have the tenant_uuid before the token in the kwargs

Reply:
fixed thanks